include/mfmg/dealii/amge_host.templates.hpp

```diff
@@ -76,10 +48,48 @@ struct NoOp
     std::ignore = src;
     std::ignore = dst;
     // Raise an error to make sure nobody uses this by inadvertance
-    throw std::runtime_error("should never get here");
+    throw std::logic_error("should never get here");
   }
 };
-#endif
+
+struct NotAMatrix
```

Comment:
`NotAMatrix` means nothing to me. Any more informative name?

Reply:
It is not a matrix.  It demonstrates it works without a matrix.  What other name would you like better?

Reply:
This `NotAMatrix` does not work without a matrix, and in fact stores the matrix explicitly inside itself. Maybe `MatrixBasedOperator`, or `MatrixWrapOperator`?